File: java/Greetings/Greetings.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.file.StandardOpenOption;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

class Greetings
{
    private static final Path FILENAME = Paths.get("greetings.txt");

    public static void main(String[] a_args) throws IOException
    {
        System.out.print("Hello, do you have a greeting for your coworkers? (Press enter to just see others' greetings)\n");
        System.out.print(">>> ");

        final String newLine;
        try (final Scanner in = new Scanner(System.in))
        {
            newLine = in.nextLine();
        }

        if (ContainsShouting(newLine))
        {
            System.out.print("Please use moderate tones...\n");
            return;
        }

        System.out.print("Here are some greetings from others...\n");
        ShowGreetings(FILENAME);

        if (!newLine.isEmpty())
        {
            Files.write(FILENAME, Arrays.asList(newLine), new StandardOpenOption[]{ StandardOpenOption.CREATE, StandardOpenOption.APPEND });
        }
    }

    private static void ShowGreetings(Path a_filename)
    {
        try
        {
            List<String> lines = Files.readAllLines(a_filename);
            for (String line : lines)
            {
                System.out.print("* ");
                System.out.println(line);
            }
        }
        catch (IOException ex)
        {
            // ignore
        }
    }

    private static boolean ContainsShouting(String a_greeting)
    {
        int contigousUpperCase = 0;
        for (char c : a_greeting.toCharArray())
        {
            contigousUpperCase = Character.isUpperCase(c) ? contigousUpperCase + 1 : 0;
            if (3 == contigousUpperCase)
            {
                return true;
            }
        }

        return false;
    }
}

File: python/Greetings/greetings.py
def ContainsShouting(a_greeting):
	contigousUpperCase = 0;
	for c in a_greeting:
		contigousUpperCase = contigousUpperCase + 1 if c.isupper() else 0
		if 3 == contigousUpperCase:
			return True

	return False

def ShowGreetings(a_filename):
	try:
		with open(a_filename, 'r') as file:
			lines = file.readlines()
			for line in lines:
				print('* ', line, end='')
	except:
		pass

FILENAME = 'greetings.txt'

def Greetings():
	print("Hello, do you have a greeting for your coworkers? (Press enter to just see others' greetings)")
	print(">>>", end=' ')

	newLine = input()

	if ContainsShouting(newLine):
		print("Please use moderate tones...")
		return None

	print("Here are some greetings from others...")
	ShowGreetings(FILENAME)

	if len(newLine) != 0:
		with open(FILENAME, 'a') as file:
			file.writelines([newLine, '\n'])

Greetings()

